docs/architecture.md

```diff
@@ -112,76 +112,206 @@ The redux store is composed of two `slices` : `documents` and `queries`:
-(for now the work is done in extractAndMergeDocument() method). 
 
-So if your app is linked to the documents store via getDocumentFromStore() for instance
-your app will have the updated value. 
+#### Auto query update
 
-`queries` has an autoUpdater mechanism that does something we can explain this way:
-- it takes the mutated documents (newly created or updated) 
-- it converts the initial `query` to a "`js predicate`" (thanks to the [sift library](https://github.com/crcn/sift.js/))
-- For each query already in the `slice` it runs this `js predicate` and detects if the query
-is concerned by the mutation
-- If the query is concerned, then it checks if it has to remove / add the id of the `mutated` 
-document
-(for now the work is done mainly in queries.js/updateData())
+Since there is a link between queries and documents through the ids list in each query object, we need to be able to deal with changes occuring in the documents collection, i.e. when documents are added or deleted.
 
-So in our previous example our `todoslist` is concerned by the addition of the new todo, then 
-the `id` is added to `todolist` data, then the component linked will be refreshed with this new 
-document.
+For this, cozy-client has a mechanism called "query updater": each time a query is run, all the queries in the store are re-evaluated directly in the store.
+That means that for each query, we take its definition, convert it into a mango predicate (see [here](#in-memory-queries-evaluation) for details), and run it against the store's document collection.
+For each query result, we check if there is any change in the known ids list, and update it accordingly.
 
-### About updates
+Thanks to this, we ensure that all the queries are always up-to-date, and apps can be effectively rendered thanks to the [react integration](/docs/react-integration.md).
 
-As already mentioned, when a mutated document already exists in the store throught its `_id`, it is updated with the new content. However, this update never removes existing fields, it only merges or adds new fields. This is typically useful when two queries retrieve the same document, but not with all the fields. 
+
+#### Fetch policy and query naming
+
+The fetch policy is a simple mechanism useful to prevent excessive fetching. 
+It is optionnaly defined for each query and define how many time should be elapsed since the last query execution before running again the query.
+
+In this example we query the checked todos, only if the last execution occured more than 10 minutes ago:
+```js
+const queryDef = Q('io.cozy.todos').where(checked: true)
+const queryOptions = {
+  as: 'io.cozy.todos/checked/true', // query unique name, to find it in the store
+  fetchPolicy: CozyClient.fetchPolicies.olderThan(10 * 60 * 1000) // fetch policy, for 10min 
+}
+await client.query(queryDef, queryOptions)
+```
+
+If the query had been run before the 10 minutes, an early return will happen and nothing will be updated in the store.
```

Comment:
this exemple is not self speaking. 

We do generally: 
```js
const queryDef = Q('io.cozy.todos').where(checked: true)
const queryOptions = {
  as: 'io.cozy.todos/checked/true', // query unique name, to find it in the store
  fetchPolicy: CozyClient.fetchPolicies.olderThan(10 * 60 * 1000) // fetch policy, for 10min 
}
const result = await client.query(queryDef, queryOptions)
```

You say that an early return will happen, ok, so what is the result of `result`?

Reply:
> We do generally:

you seem to have copy/pasted the the same code?

I changed the example to better demonstrate what happens, is it clearer?

Reply:
I just added the `result` assignation of the `client.query`. Because if you are not using the useQuery hooks, for instance you are in a node context or within a React Class Component, then if you run a `client.query` with a fetchPolicy and that fetchPolicy says "I don't need to fetch", then the result will be empty. 

https://github.com/cozy/cozy-client/issues/931